src/Compiler/Service/ServiceParseTreeWalk.fsi

```diff
@@ -294,26 +271,27 @@ module public ParsedInput =
     /// </summary>
     /// <param name="position">The position in the input file down to which to apply the function.</param>
     /// <param name="chooser">The function to apply to each node and its context to derive an optional value.</param>
-    /// <param name="parsedInput">The AST to search.</param>
+    /// <param name="ast">The AST to search.</param>
     /// <returns>The first value for which the function returns <c>Some</c>, or <c>None</c> if no matching node is found.</returns>
-    val tryPickDownTo:
-        position: pos ->
-        chooser: (SyntaxVisitorPath -> SyntaxNode -> 'T option) ->
-        parsedInput: ParsedInput ->
-            'T option
+    /// <example>
+    /// <code lang="fsharp">
+    /// let range =
+    ///     parseResults.ParseTree.Contents |> Ast.tryPick pos (fun _path node ->
+    ///       match node with
+    ///       | SyntaxNode.SynExpr (SynExpr.InterpolatedString (range = range)) when rangeContainsPos range pos -> Some range
+    ///       | _ -> None)
+    /// </code>
+    /// </example>
+    val tryPick: position: pos -> chooser: (SyntaxVisitorPath -> SyntaxNode -> 'T option) -> ast: Ast -> 'T option
 
-    /// <summary>
-    /// Like fold, but the walker function may optionally specify which child nodes to walk in what order.
-    ///
-    /// If the walker function specifies <c>None</c> for the child nodes, the node's children are traversed in the default way.
-    /// If the walker function specifies <c>Some []</c> for the child nodes, none of the node's children are traversed.
-    /// </summary>
-    /// <param name="walker">The function to use to update the state given each node and its context.</param>
-    /// <param name="state">The initial state.</param>
-    /// <param name="parsedInput">The AST to fold over.</param>
-    /// <returns>The final state.</returns>
-    val walk:
-        walker: ('State -> SyntaxVisitorPath -> SyntaxNode -> 'State * SyntaxNode list option) ->
-        state: 'State ->
-        parsedInput: ParsedInput ->
-            'State
+/// <summary>
+/// Holds extension methods for <see cref="T:FSharp.Compiler.Syntax.ParsedInput"/>.
+/// </summary>
+[<AutoOpen>]
+module ParsedInputExtensions =
+    type ParsedInput with
+
+        /// <summary>
+        /// Gets the parsed file's contents as forest of <see cref="T:FSharp.Compiler.Syntax.SyntaxNode"/>s.
+        /// </summary>
+        member Contents: Ast
```

Comment:
Maybe go with `member Nodes` not to clash with `ParsedImplFileInput(contents = contents)`?

Reply:
Hmm, they _are_ arguably the same "contents" as the underlying `SynModuleOrNamespace{Sig}`s; they're just unified as `SyntaxNode`s.